gadget/device_linux.go

```diff
@@ -54,11 +55,23 @@ func FindDeviceForStructure(ps *PositionedStructure) (string, error) {
 	var found string
 	var match string
 	for _, candidate := range candidates {
-		if !osutil.FileExists(candidate) {
-			continue
+		st, err := os.Lstat(candidate)
```

Comment:
Can we skip lstat here and simply do `target, err := filepath.EvalSymlinks(candidate)`, will EvalSymlink error out if candidate is not a symlink?

Reply:
From earlier testing, EvalSymlink would not error out. That's why I added an explicit check for a symlink.